share/wake/html/timeline.html

```diff
@@ -0,0 +1,795 @@
+            let top_y = item.parent.top + item.parent.height - item.top - item.height;
+            return {
+                left: left_x,
+                top: top_y,
+                right: left_x + item.width,
+                bottom: top_y + item.height,
+                mid_x: left_x + item.width / 2,
+                mid_y: top_y + item.height / 2,
+                width: item.width,
+                height: item.height
+            }
+        }
+
+
+        addArrow (dep) {
+            this._dependency.push(dep);
+            this._createPath();
+            this._timeline.redraw();
+        }
+
+        getArrow (id) {
+            for (let i = 0; i < this._dependency.length; i++) {
+                if (this._dependency[i].id == id) {
+                    return this._dependency[i];
+                }
+            }
+            return null;
+        }
+
+        //Función que recibe el id de una flecha y la elimina.
+        removeArrow(id) {
+            for (let i = 0; i < this._dependency.length; i++) {
+                if (this._dependency[i].id == id) var index = i;
+            }
+
+            //var list = document.getElementsByTagName("path"); //FALTA QUE ESTA SELECCION LA HAGA PARA EL DOM DEL TIMELINE INSTANCIADO!!!!
+            var list = document.querySelectorAll("#" +this._timeline.dom.container.id +" path");
+
+            this._dependency.splice(index, 1); //Elimino del array dependency
+            this._dependencyPath.splice(index, 1); //Elimino del array dependencyPath
+
+            list[index + 1].parentNode.removeChild(list[index + 1]); //Lo elimino del dom
+        }
+
+        //Función que recibe el id de un item y elimina la flecha.
+        removeArrowbyItemId(id) {
+            let listOfRemovedArrows = [];
+            for (let i = 0; i < this._dependency.length; i++) {
+                if ( (this._dependency[i].id_item_1 == id) || (this._dependency[i].id_item_2 == id) ) {
+                    listOfRemovedArrows.push(this._dependency[i].id);
+                    this.removeArrow(this._dependency[i].id);
+                    i--;
+                }
+            }
+            return listOfRemovedArrows;
+        }
+    }
+</script>
+</body>
+</html>
+
+<script type="module">
+    /* Copyright 2022 SiFive, Inc.
+     *
+     * Licensed under the Apache License, Version 2.0 (the "License");
+     * you may not use this file except in compliance with the License.
+     * You should have received a copy of LICENSE.Apache2 along with
+     * this software. If not, you may obtain a copy at
+     *
+     *    https://www.apache.org/licenses/LICENSE-2.0
+     *
+     * Unless required by applicable law or agreed to in writing, software
+     * distributed under the License is distributed on an "AS IS" BASIS,
+     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     * See the License for the specific language governing permissions and
+     * limitations under the License.
+     */
+
```

Comment:
A general comment about the organization of this file: This file is quite long and does several different kinds of things, and it would help with readability of similar "kinds" of things are grouped together. Some possible groupings you may want to have include:

1. Things that deal with HTML or DOM operations
2. Things that deal with setting up the `vis.js` Timeline objects
3. The JobNode class and any functions/methods that operate on JobNodes
4. Your graph operations, such as DFS or topological sorting
5. Other miscellaneous setting of initial state
6. Functions that deal with updating the state of the application in response to user-initiated events


A common architectural pattern for organizing user interfaces is [MVC](https://developer.mozilla.org/en-US/docs/Glossary/MVC) (Model-View-Controller), where 1) and 2) are mostly the View; 3), 4), and 5) are mostly the Model, and 6) is the Controller.

Reply:
Grouped.